### PR TITLE
bzimage command and package support for same

### PR DESCRIPTION
**Pull request by rminnich**
bzImage dump bzImage
bzImage initramfs bzImage initramfs newbzImage

In support of this, add an init.cpio for testing.
While we're at it, add the stuff we use to make it.

In conjunction with the kernel in testdata/bzImage, this creates
a working kernel with an initramfs.

One limit: we require that the kernel have a fixed load address,
so we can set that address in the LinuxHeader (bootparams).
I don't think this is a big deal. KASLR is really not needed
for a firmware kernel.

CONFIG is to create testdata/bzImage.

We also don't support placing the kernel and initramfs
above the 4G limit. To do that we'd need to add code
setting
        ExtRamdiskImage
        ExtRamdiskSize
in the LinuxHeader. That's just crazy.

Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

File: cmds/bzImage/bzImage_test.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+func TestSimple(t *testing.T) {
+	tmpDir, err := ioutil.TempDir("", "bzImage")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(tmpDir)
+	// adjust the args for the output test.
+	// Create a bogus path in our tmpDir
+	tests[0].args[3] = filepath.Join("zz", "zz", "zz")
```
Comment (rjoleary) [inline cmds/bzImage/bzImage_test.go]: If you move the table `b` to be a local variable defined here, you won't need to swap out the placeholder values.

File: cmds/bzImage/bzImage.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+// bzImage is used to modify bzImage files. It will
+// read the image in, apply any operators we can dream
+// up, then write it out.
+//
+// Synopsis:
+//     bzImage [dump <file>] | [initramfs input-bzimage initramfs output-bzimage]
+//
+// Description:
+//	Read a bzImage in, change it, write it out, or print info.
+//
```
[the remark this answers is not in the document]
Comment (rminnich) [reply cmds/bzImage/bzImage.go]: done

File: cmds/bzImage/bzImage.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2012-2018 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// bzImage is used to modify bzImage files. It will
```
[the remark this answers is not in the document]
Comment (rminnich) [reply cmds/bzImage/bzImage.go]: done

File: cmds/bzImage/bzImage_test.go [hunk 4]
```diff
@@ -0,0 +1,182 @@
+// Copyright 2017-8 the u-root Authors. All rights reserved
```
[the remark this answers is not in the document]
Comment (rminnich) [reply cmds/bzImage/bzImage_test.go]: done

File: cmds/bzImage/bzImage.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+// read the image in, apply any operators we can dream
+// up, then write it out.
+//
+// Synopsis:
+//     bzImage [dump <file>] | [initramfs input-bzimage initramfs output-bzimage]
```
[the remark this answers is not in the document]
Comment (rminnich) [reply cmds/bzImage/bzImage.go]: mmm ... I don't like it. The reason is that anything this gets piped to will have to handle embedded newlines. Best to keep it simple IMHO for cases where a program is reading stderr. But I'm willing to listen to reason.

File: pkg/bzimage/bzImage.go [hunk 6]
```diff
@@ -145,14 +145,14 @@ func (h *LinuxHeader) Show() []string {
...
+		n := fmt.Sprintf("%s", val.Type().Field(i).Name)
 		switch k {
 		case reflect.Slice:
-			s = append(s, fmt.Sprintf("%s:%02x", n, v))
+			s = append(s, fmt.Sprintf("%s:0x%02x", n, v))
 		case reflect.Bool:
 			s = append(s, fmt.Sprintf("%s:%v", n, v))
 		default:
-			s = append(s, fmt.Sprintf("%s:%02x", n, v))
+			s = append(s, fmt.Sprintf("%s:0x%02x", n, v))
```
[the remark this answers is not in the document]
Comment (rminnich) [reply pkg/bzimage/bzImage.go]: done

File: cmds/bzImage/bzImage_test.go (hunk 1 above)
Comment (rminnich) [reply cmds/bzImage/bzImage_test.go]: done

File: cmds/bzImage/bzImage_test.go [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+	}
+	defer os.RemoveAll(tmpDir)
+	// adjust the args for the output test.
+	// Create a bogus path in our tmpDir
+	tests[0].args[3] = filepath.Join("zz", "zz", "zz")
+	// Use a real path
+	tests[1].args[3] = filepath.Join(tmpDir, "zz")
+	// Table-driven testing
+	for _, tt := range tests {
+		c := testutil.Command(t, tt.args...)
```
[the remark this answers is not in the document]
Comment (rminnich) [reply cmds/bzImage/bzImage_test.go]: oh cool! thanks for that! done.

File: pkg/bzimage/bzImage.go [hunk 8]
```diff
@@ -145,14 +145,14 @@ func (h *LinuxHeader) Show() []string {
 	for i := 0; i < val.NumField(); i++ {
 		v := val.Field(i)
 		k := reflect.ValueOf(v).Kind()
-		n := fmt.Sprintf("%s:", val.Type().Field(i).Name)
+		n := fmt.Sprintf("%s", val.Type().Field(i).Name)
 		switch k {
 		case reflect.Slice:
-			s = append(s, fmt.Sprintf("%s:%02x", n, v))
+			s = append(s, fmt.Sprintf("%s:0x%02x", n, v))
```
[the remark this answers is not in the document]
Comment (rminnich) [reply pkg/bzimage/bzImage.go]: done

**Pull request merged**